monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/rest/delegates/heartbeat/HeartBeatDelegate.java

```diff
@@ -80,63 +83,80 @@ private boolean isNodeRegistered(HeartbeatObject heartbeat) {
         return (null != timestamp && !timestamp.isEmpty());
     }
 
-    private boolean updateHeartbeat(HeartbeatObject heartbeat) {
+    private boolean updateHeartbeat(HeartbeatObject heartbeat) throws ManagementApiException {
         if (logger.isDebugEnabled()) {
             logger.debug("Updating heartbeat information of node " + heartbeat.getNodeId() + " in group : " +
                       heartbeat.getGroupId());
         }
-        return dataManager.updateHeartbeat(heartbeat);
+        try {
+            return dataManager.updateHeartbeat(heartbeat);
+        } catch (DashboardServerException e) {
+            throw new ManagementApiException (e.getMessage(), 500);
```

Comment:
With this the exception stack trace will be lost?

Reply:
Added e.getCause() now.